libsolidity/analysis/StaticAnalyzer.cpp

```diff
@@ -231,6 +232,45 @@ bool StaticAnalyzer::visit(InlineAssembly const& _inlineAssembly)
 	return true;
 }
 
+bool StaticAnalyzer::visit(BinaryOperation const& _operation)
+{
+	if (
+		_operation.rightExpression().annotation().isConstant &&
+		(_operation.getOperator() == Token::Div || _operation.getOperator() == Token::Mod)
+	)
+		if (auto result = dynamic_pointer_cast<RationalNumberType const>(
+			ConstantEvaluator(m_errorReporter).evaluate(_operation.rightExpression())
+		))
+			if (result->literalValue(nullptr) == u256(0))
+				m_errorReporter.typeError(
+					_operation.location(),
+					"Division by zero."
```

Comment:
I think this should be more specific based on the token: `Modulo zero` or `Division by zero`.